bench-tx/src/main.rs

```diff
@@ -13,15 +13,16 @@ use miden_objects::{
     assets::{Asset, FungibleAsset},
     crypto::{dsa::rpo_falcon512::SecretKey, rand::RpoRandomCoin},
     notes::NoteType,
+    testing::notes::AssetPreservationStatus,
     transaction::TransactionArgs,
     Felt,
 };
 use miden_tx::{
-    auth::BasicAuthenticator, testing::data_store::MockDataStore, utils::Serializable,
```

Comment:
Removed `MockDataStore` in favor of using `TransactionContextBuilder` as a `DataStore`. This removes one level of indirection, and makes the state of the tests more explicit (it is now part of the test body, instead of being hidden in the `MockDataStore` constructors)